ce/ce/amf/demands.ts

```diff
@@ -298,33 +290,27 @@ export class Unless extends DemandBlock implements AlternativeFulfillment {
           }
         })) {
           // we found something that looks promising.
-          let filtered = <any>{ $0: item };
-          this.setData(filtered);
-          if (this.run) {
+          this.discoveredData = { $0: item.toString() };
+          const run = this.run?.replace('$0', item.toString());
 
-            const commandline = cmdlineToArray(this.run.replace('$0', item.toString()));
-            const result = await execute(resolve(commandline[0]), commandline.slice(1));
-            if (result.code !== 0) {
+          if (run) {
+            const result = await this.runCmd(run);
+            if (!result) {
               continue;
             }
 
-            filtered = filter(this.select || '', result.log);
-            filtered.$0 = item;
-
-            // if we have a match expression, let's check it.
-            if (this.matches && !safeEval(this.matches, filtered)) {
-              continue; // not a match, move on
-            }
-
-            // it did match, or it's just presence check
+            this.discoveredData['$0'] = item.toString();
```

Comment:
I don't understand why this line is important given 293; isn't this just setting `this.discoveredData['$0']` to exactly what it already is?